dashboard/modules/reporter/tests/test_reporter.py

```diff
@@ -697,5 +696,127 @@ def _generate_worker_key(self, proc):
             agent_mock.kill()
 
 
+TASK = {
+    "task_id": "32d950ec0ccf9d2affffffffffffffffffffffff01000000",
+    "attempt_number": 0,
+    "node_id": "ffffffffffffffffffffffffffffffffffffffff01000000",
+}
+
+
+def test_get_task_traceback_running_task(shutdown_only):
+    """
+    Verify that we throw an error for a non-running task.
+
+    """
+    # The sleep is needed since it seems a previous shutdown could be not yet
+    # done when the next test starts. This prevents a previous cluster to be
+    # connected the current test session.
+
+    time.sleep(5)
+    address_info = ray.init()
+    webui_url = format_web_url(address_info["webui_url"])
+
+    @ray.remote
+    def f():
+        pass
+
+    @ray.remote
+    def long_running_task():
+        print("Long-running task began.")
+        time.sleep(1000)
+        print("Long-running task completed.")
+
+    ray.get([f.remote() for _ in range(5)])
+
+    task = long_running_task.remote()
+
+    params = {
+        "task_id": task.task_id().hex(),
+        "attempt_number": 0,
+        "node_id": ray.get_runtime_context().node_id.hex(),
+    }
+
+    def verify():
+        resp = requests.get(f"{webui_url}/task/traceback", params=params)
+        print(f"resp.text {type(resp.text)}: {resp.text}")
+
+        assert "Process" in resp.text
+        return True
+
+    wait_for_condition(verify, timeout=20)
+
+
+def test_get_task_traceback_non_running_task(shutdown_only):
+    """
+    Verify that we throw an error for a non-running task.
+    """
+
+    # The sleep is needed since it seems a previous shutdown could be not yet
+    # done when the next test starts. This prevents a previous cluster to be
+    # connected the current test session.
+
+    time.sleep(5)
+    address_info = ray.init()
+    webui_url = format_web_url(address_info["webui_url"])
+
+    @ray.remote
+    def f():
+        pass
+
+    ray.get([f.remote() for _ in range(5)])
+
+    params = {
+        "task_id": TASK["task_id"],
+        "attempt_number": TASK["attempt_number"],
+        "node_id": TASK["node_id"],
+    }
+
+    # Make sure the API works.
+    def verify():
+        with pytest.raises(requests.exceptions.HTTPError) as exc_info:
+            resp = requests.get(f"{webui_url}/task/traceback", params=params)
+            resp.raise_for_status()
+        assert isinstance(exc_info.value, requests.exceptions.HTTPError)
+        return True
+
+    wait_for_condition(verify, timeout=10)
+
+
+def test_get_cpu_profile_non_running_task(shutdown_only):
+    """
+    Verify that we throw an error for a non-running task.
+    """
+
+    # The sleep is needed since it seems a previous shutdown could be not yet
+    # done when the next test starts. This prevents a previous cluster to be
+    # connected the current test session.
+
+    time.sleep(5)
```

Comment:
all the sleeps might not be needed if you are doing `ray.init` versus running `ray start/stop`.

Reply:
"Do you know why 'is' is not needed? I noticed that you called 'sleep' before 'ray.init' in this GitHub pull request: [Link](https://github.com/ray-project/ray/pull/38440/files#diff-35a5c33a4d0f5fe42fec562ad514973afd2a581b0738b6b8c8a9ae81c4123b05R650).

I read the documentation for 'ray.init':

> This method handles two cases: either a Ray cluster already exists and we just attach this driver to it, or we start all of the processes associated with a Ray cluster and attach to the newly started cluster.

It seems that 'ray.init' will start a cluster under the hood if no cluster exists."

Reply:
does it fail if you don't sleep here? 

ray start and ray.init have slightly different semantics. I called ray start in the example you linked.

Reply:
Got it, thanks for the info. I noticed you did have a fixture `call_ray_start` in the test case

Reply:
I will check whether it will fail later